### ISSUE TITLE
Tidy up project and gradle tasks

### ISSUE THREAD

**Issue by bric3** [mockito/mockito#766]
I just looked at the [build.gradle](https://github.com/spring-projects/spring-framework/blob/master@{2016-11-18}/build.gradle) file from spring, and it looks a lot more cleaner.

I propose to revise how our project structure is organised and how gradle is structured as well.

* spring has a [common configuration](https://github.com/spring-projects/spring-framework/blob/master@%7B2016-11-18%7D/build.gradle#L29) for all projects / subprojects, while ours seems related to the current (mockito-core) project only
* spring has unique place where [root project task are configured](https://github.com/spring-projects/spring-framework/blob/master@%7B2016-11-18%7D/build.gradle#L1183)
* spring has a way to express (dependencies versions](https://github.com/spring-projects/spring-framework/blob/master@%7B2016-11-18%7D/build.gradle#L33-L100) while our dependencies versions get spread out in various files (javadoc, osgi, etc.)
* spring has properly [configured compiled tasks](https://github.com/spring-projects/spring-framework/blob/master@%7B2016-11-18%7D/build.gradle#L146-L157) while ours are at the root of the file.
* spring can configure [each sub-project](https://github.com/spring-projects/spring-framework/blob/master@%7B2016-11-18%7D/build.gradle#L323) independently
* spring [project/directory structure is flat](https://github.com/spring-projects/spring-framework/tree/master@%7B2016-11-18%7D) while ours has subproject in the main project (which raises confusion about mockito-core, root project, etc.)

**Comment by szczepiq** [comment]
Absolutely. Great ideas. Thank you for looking into this!!!

**Comment by bric3** [comment]
If you're okay I'd like to work on this one. I really struggle with gradle sometime, but It may prove a good thing if working on this. Of course when the branch will be pushed anyway can help on this.

**Comment by TimvdLippe** [comment]
@bric3 Yes go ahead :smile:

**Comment by szczepiq** [comment]
>If you're okay I'd like to work on this one.

I'm absolutely ok and super excited that it will get cleaned. Thanks a lot for being proactive and keeping the house tidy!!!

**Comment by TimvdLippe** [comment]
We are in a lot better state right now. Most of the subproject gradle files are clean, or as clean as they can be. Most of the logic is now in `java-library.gradle`. Other than that, we extracted our common dependency versions in `dependencies.gradle` per #1075 Lastly most of the release logic has been extracted into Shipkit.

All in all, I think most of the issues have been addressed, thus I am closing this issue.
